lasagne/layers/recurrent.py

```diff
@@ -200,7 +215,7 @@ def step(input_dot_W_n, hid_previous, *args):
                 original_hid_pre,
                 replace=dict(zip(original_params, new_params)))
 
-            new_hid_pre += input_dot_W_n
+            new_hid_pre += compute_in_to_hid(input_dot_W_n)
```

Comment:
I think we should rename to the input to `input_n`? 

I prefer if/else instead of the lambda functions. What do you think?